crates/next-api/src/app.rs

```diff
@@ -880,21 +880,33 @@ impl AppEndpoint {
                     None
                 };
 
-                let client_dynamic_imports = collect_next_dynamic_imports(
-                    client_references
-                        .await?
-                        .client_references
-                        .iter()
-                        .filter_map(|r| match r.ty() {
-                            ClientReferenceType::EcmascriptClientReference(entry) => Some(entry),
-                            ClientReferenceType::CssClientReference(_) => None,
-                        })
-                        .map(|entry| async move { Ok(Vc::upcast(entry.await?.ssr_module)) })
-                        .try_join()
-                        .await?,
-                    Vc::upcast(this.app_project.client_module_context()),
-                )
-                .await?;
+                let mut client_references_by_server_comp = IndexMap::new();
+                for r in &client_references.await?.client_references {
+                    if let ClientReferenceType::EcmascriptClientReference(entry) = r.ty() {
+                        client_references_by_server_comp
+                            .entry(r.server_component())
+                            .or_insert_with(Vec::new)
+                            .push(Vc::upcast::<Box<dyn Module>>(entry.await?.ssr_module));
+                    }
+                }
```

Comment:
I think it would make sense to move that logic into `client_reference_graph()` and create the map there

Reply:
Done, I also (had to) add a `Vc` around each vector value